### PR TITLE
Update python 3.10

### PR DESCRIPTION
**Pull request by Yuyuutsu**
### What is the context of this PR?
This PR is upgrading the python version from 3.9 to 3.10.7 to ensure we have the latest features. Docker and pipfile also updated.  
Other updates: 
- Black package upgraded to latest (unpinned)
- Files were also reformatted to meet PyLint= 2.15.3, and Black=22.8.0 standards.
- Pipenv updated in YAML and DockerFile to 2022.9.24
- Read.me updated to install pyenv via git instead of brew
- New .pylintrc generated 

### How to review 
- Test if pyenv installation guide on the `README.md` works 
- Check if tests and Makefile scripts are working fine

### CODE REVIEW

**Comment by Yuyuutsu** [comment]
[Upgrade non-runner Python repos to 3.10 (2 SP)](https://trello.com/c/B9pw2uPW/5229-upgrade-non-runner-python-repos-to-310-2-sp)

**Comment by Yuyuutsu** [comment]
[Upgrade non-runner Python repos to 3.10 (2 SP)](https://trello.com/c/B9pw2uPW/5229-upgrade-non-runner-python-repos-to-310-2-sp)

**Comment by Yuyuutsu** [comment]
[Upgrade non-runner Python repos to 3.10 (2 SP)](https://trello.com/c/B9pw2uPW/5229-upgrade-non-runner-python-repos-to-310-2-sp)

File: Pipfile.lock [hunk 1]
```diff
@@ -231,85 +256,117 @@
...
-                "sha256:3b80836aa6d1feeaa108e046da6423ab8f6ceda6468545ae8d02d9d58d18818a"
+                "sha256:51c75c4126074b472f746a24399ad32f6053d1b34b68d2fa41e558e6f4a98719",
+                "sha256:607c53218732647dff4acdfcd50cb62615cedf612e72d1724fb1a0cc6405b378"
             ],
-            "markers": "python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3'",
-            "version": "==1.10.0"
+            "markers": "python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3, 3.4'",
+            "version": "==1.11.0"
         },
         "pylint": {
```
Comment (petechd) [inline Pipfile.lock:363]: We need to decide if we pin the version or update the `.pylintrc` file.

File: .pylintrc [hunk 2]
```diff
@@ -41,7 +41,7 @@ extension-pkg-whitelist=
 # Pylint and this flag can prevent that. It has one side effect, the resulting
 # AST will be different than the one from reality. This option is deprecated
 # and it will be removed in Pylint 2.0.
-optimize-ast=no
+#optimize-ast=no
```
Comment (MebinAbraham) [inline .pylintrc:44]: How comes this changed?

File: .pylintrc [hunk 3]
```diff
@@ -87,7 +86,7 @@ output-format=text
 # command line instead of printing them on stdout. Reports (if any) will be
 # written in a file name "pylint_global.[txt|html]". This option is deprecated
 # and it will be removed in Pylint 2.0.
-files-output=no
+#files-output=no
```
Comment (MebinAbraham) [inline .pylintrc:89]: How comes this changed?

File: Pipfile [hunk 4]
```diff
@@ -15,12 +15,12 @@ jsonpath-rw = "*"
 pylint = "*"
 pytest = "*"
 pylint-quotes = "*"
-black = "==19.10b0"
+black = "==22.8.0"
```
Comment (MebinAbraham) [inline Pipfile:18]: Worth unpinning.

File: .pylintrc (hunk 3 above)
Comment (Yuyuutsu) [reply .pylintrc:89]: files-ouput was removed in PyLint [1.7](https://pylint.pycqa.org/en/latest/whatsnew/1/1.7/summary.html#removed-changes)

File: Pipfile (hunk 4 above)
Comment (Yuyuutsu) [reply Pipfile]: Done 👍

File: .pylintrc (hunk 2 above)
Comment (Yuyuutsu) [reply .pylintrc]: Generated new .pylintrc and updated any overides we had

File: Pipfile.lock (hunk 1 above)
Comment (Yuyuutsu) [reply Pipfile.lock:361]: Generated new .pylintrc and updated any overides we had

**Review by berroar — APPROVED** [review]

**Review by petechd — APPROVED** [review]

**Pull request merged**